### PR TITLE
[JW8-10957] Use attached flag in setActiveItem while waiting for asyncActiveItem

### PR DESCRIPTION
**Pull request by pajong**
### This PR will...
- Use `attached` flag in `mediaController` while waiting for the `asyncActiveItem` to block events

### Why is this Pull Request needed?
- Calling `mediaController.off` results all the events until `this._setActiveMedia(mediaController);` call, especially when we are re-using the same `mediaController` from the previous item
- `mediaType` event is fired when setting a new activeItem to `mediaController`
`mediaController.activeItem = playlistItem;`
-- `mediaType` event is lost if we use `mediaController.off`
-- Using `detach` to queue up the events and later flushing with `attach` solves the issue

### Are there any points in the code the reviewer needs to double check?
- Tried using `mediaController.attach()` and `mediaController.detach()`, but it was causing other regressions since they have other function calls (`setPlaybackRate` and ` provider.attachMedia()`)

#### Addresses Issue(s):

JW8-10957

### Checklist
- [ ] Jenkins builds and unit tests are passing
- [ ] I have reviewed the automated results

### CODE REVIEW

File: src/js/program/program-controller.js [hunk 1]
```diff
@@ -122,6 +122,8 @@ class ProgramController extends Events {
                     // We can synchronously reuse the current mediaController
                     // Reinitialize the mediaController with the new item, allowing a new playback session
                     mediaController.activeItem = playlistItem;
+                    mediaController.attached = true;
+                    mediaController.eventQueue.flush();
```
Comment (robwalch) [inline src/js/program/program-controller.js]: Can these go in `_setActiveMedia`? It's called from a couple places and will always need to be attached.

**Comment by pajong** [comment]
test this please

File: src/js/program/program-controller.js [hunk 2]
```diff
@@ -78,7 +78,7 @@ class ProgramController extends Events {
         // Reset the mediaModel now to synchronously "cancel" play promise callbacks
         // that check `mediaModel === model.mediaModel`
         if (mediaController) {
-            mediaController.off();
+            mediaController.attached = false;
```
Comment (robwalch) [inline src/js/program/program-controller.js]: Just remove off, and don't use `attached`. We can put it back and add listeners earlier if we need to.
